src/src/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -595,33 +589,29 @@ private AuthenticationResult processTokenResponse(HttpWebResponse webResponse) {
                 correlationIdInHeader = listOfHeaders.get(0);
             }
         }
-
-        if (webResponse.getStatusCode() == HttpURLConnection.HTTP_OK
-                && webResponse.getBody() != null && webResponse.getBody().length > 0) {
-            // invalid refresh token calls has error related items in the body.
-            // Status is 400 for those.
-            try {
-                String jsonStr = new String(webResponse.getBody());
-                extractJsonObjects(responseItems, jsonStr);
-                result = processUIResponseParams(responseItems);
-            } catch (final JSONException ex) {
-                // There is no recovery possible here, so
-                // catch the
-                // generic Exception
-                Logger.e(TAG, ex.getMessage(), "", ADALError.SERVER_INVALID_JSON_RESPONSE, ex);
-                result = new AuthenticationResult(JSON_PARSING_ERROR, ex.getMessage(), null);
-            }
+        
+        final boolean isResponseBodyEmpty = webResponse.getBody() == null || webResponse.getBody().length == 0;
```

Comment:
consider putting brackets around each comparison. this way it will be harder to introduce a bug.